### PR TITLE
Update the std::thread docs and clarify that panics can nowadays be caught

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -27,11 +27,15 @@
...
 //! Fatal logic errors in Rust cause *thread panic*, during which
 //! a thread will unwind the stack, running destructors and freeing
-//! owned resources. Thread panic is unrecoverable from within
-//! the panicking thread (i.e. there is no 'try/catch' in Rust), but
-//! the panic may optionally be detected from a different thread. If
-//! the main thread panics, the application will exit with a non-zero
-//! exit code.
+//! owned resources. While not meant as a 'try/catch' mechanism, panics
+//! in Rust can nonetheless be caught with
+//! ['catch_unwind'](../../std/panic/fn.catch_unwind.html) and recovered
```
Comment (kennytm) [inline src/libstd/thread/mod.rs]: Better use backquotes (`` `…` ``) here (same for the `resume_unwind` below).

Comment (sdroege) [reply src/libstd/thread/mod.rs]: Good point, thanks. Will update in a bit
The wording is otherwise sounding OK?

Comment (steveklabnik) [reply src/libstd/thread/mod.rs]: Another suggestion :) This assumes that they can *always* be caught, but with panic=abort, they cannot still.

Comment (sdroege) [reply src/libstd/thread/mod.rs]: Also a very good point :) Thanks

**Review by dtolnay — APPROVED** [review]

**Pull request merged**